site/docs/starter/slash-commands.md

```diff
@@ -0,0 +1,550 @@
+    // Due to the way discord internally handles global commands, it might take up to an hour before they are available.
+    log.info(
+      "Updating Global Commands, changes should apply shortly...",
+    );
+    // Upsert will add commands, replace existing commands, and delete ones that do not exist anymore.
+    await bot.helpers.upsertApplicationCommands(globalCommands).catch(
+      log.error,
+    );
+  }
+
+  if (perGuildCommands.length && (scope === "Guild" || scope === undefined)) {
+    await bot.guilds.forEach(async (guild: Guild) => {
+      // Upsert will add commands, replace existing commands, and delete ones that do not exist anymore.
+      await upsertApplicationCommands(bot, perGuildCommands, guild.id);
+    });
+  }
+}
+
+/** Update commands for a guild */
+export async function updateGuildCommands(bot: Bot, guild: Guild) {
+  const perGuildCommands: MakeRequired<CreateApplicationCommand, "name">[] = [];
+
+  // Goes over all commands in the command map, so it's important to load the commands before running this function.
+  for (const command of commands.values()) {
+    if (command.scope) {
+      // Since we are only updating guild commands, we should only select commands with the "Guild" scope.
+      if (command.scope === "Guild") {
+        perGuildCommands.push({
+          name: command.name,
+          description: command.description,
+          type: command.type,
+          options: command.options ? command.options : undefined,
+        });
+      }
+    }
+  }
+
+  if (perGuildCommands.length) {
+    // Upsert will add commands, replace existing commands, and delete ones that do not exist anymore.
+    await upsertApplicationCommands(bot, perGuildCommands, guild.id);
+  }
+}
+// ...
+```
+
+We'll add this to our `mod.ts` to make sure that we always have the latest commands ready for users:
+
+```typescript title="mod.ts"
+// Imports
+import { logger } from "./src/utils/logger.ts";
+import {
+  dotEnvConfig,
+  enableCachePlugin, 
+  createBot, 
+  GatewayIntents, 
+  enableCacheSweepers, 
+  startBot 
+} from "./deps.ts";
+import { events } from "./src/events/mod.ts";
+// highlight-next-line
+import { updateCommands } from "./src/utils/helpers.ts";
+
+const log = logger({ name: "Main" });
+
+// Secrets
+dotEnvConfig({ export: true });
+export const BOT_TOKEN = Deno.env.get("BOT_TOKEN") || "";
+export const BOT_ID = BigInt(atob(BOT_TOKEN.split(".")[0]));
+
+log.info("Starting Bot, this might take a while...");
+
+// Create the Bot Object
+export const bot = enableCachePlugin(createBot({
+  token: BOT_TOKEN,
+  botId: BOT_ID,
+  intents: GatewayIntents.Guilds | GatewayIntents.GuildMembers | GatewayIntents.GuildMessages | GatewayIntents.DirectMessages,
+  events
+}));
+
+enableCacheSweepers(bot);
+
+if (BOT_TOKEN) {
+  // Start the bot
+  await startBot(bot);
+  // highlight-start
+  // Update all commands
+  await updateCommands(bot);
```

Comment:
I would recommend putting these two in a Promise.all() so it is slightly faster, commands can be updated without having the bot online.